app/models/mail_handler.rb

```diff
@@ -161,8 +161,8 @@ def receive_issue
     end
 
     issue = WorkPackage.new(author: user, project: project)
-    issue.safe_attributes = issue_attributes_from_keywords(issue)
-    issue.safe_attributes = { 'custom_field_values' => custom_field_values_from_keywords(issue) }
+    issue.attributes = issue_attributes_from_keywords(issue)
```

Comment:
After taking a deeper look into the current `WorkPackage#safe_attributes=` implementation I think that the change does alter the behaviour. The method implements some business logic:
1) Some attributes are removed when the work package has a parent
2) The new status is checked
3) The parent_id is removed on some cases.

In my opinion the above problem and the one with with `issue_attributes_from_keywords` boils down to an architectural decision. Is permitted_params only responsible for statically whitelisting some attributes for mass assignment or is there business logic involved? Currently it is the later as decisions are made depending e.g. on the user and for some parameters also on the state of the object the attributes are to be assigned to. 

So a simple solution might be to 
1) move the business logic in `safe_attributes=` to `PermittedParams`. You have already done that for other models.
2) Wrap the response from `issue_attributes_from_keywords` in the same permit logic to benefit from the business logic. 
This solution might be good enough for now. 

However, I find this to be a suboptimal solution. Doing that we are spreading the business logic (e.g. Model, PermittedParams, Controller, Mailer). 
We need some object, that contains the whole of the business logic. There already exist some service objects so they might be the container I am looking for. DCI or some other POR business layer between controller and model might also do the trick.